ui/app/components/tools/wrap.js

```diff
@@ -24,15 +25,31 @@ export default class ToolsWrap extends Component {
   @tracked buttonDisabled = false;
   @tracked token = '';
   @tracked wrapTTL = null;
-  @tracked wrapData = '{\n}';
+  @tracked wrapData;
   @tracked errorMessage = '';
+  @tracked showJson = true;
+  get jsonWrapStartValue() {
+    return '{\n}';
+  }
+
+  get startingValue() {
+    // must pass the third param called "space" in JSON.stringify to structure object with whitespace
+    // otherwise the following codemirror modifier check will pass `this._editor.getValue() !== namedArgs.content` and _setValue will be called.
+    // the method _setValue moves the cursor to the beginning of the text field.
+    // the effect is that the cursor jumps after the first key input.
+    return JSON.stringify({ '': '' }, null, 2);
+  }
+
+  get stringifiedWrapData() {
+    return this?.wrapData ? stringify([this.wrapData], {}) : this.jsonWrapStartValue;
+  }
 
   @action
   reset(clearData = true) {
     this.token = '';
     this.errorMessage = '';
     this.wrapTTL = null;
-    if (clearData) this.wrapData = '{\n}';
+    if (clearData) this.wrapData = '';
```

Comment:
We should keep the type consistent for `this.wrapData`. Since we use it for a falsey check above - kinda thinking we should initialize this as `null` and reset it to that for consistency here. That way if it does have a value, we know it will consistently be an object.